test/boost/mutation_reader_test.cc

```diff
@@ -1000,10 +995,9 @@ SEASTAR_TEST_CASE(reader_selector_fast_forwarding_test) {
 
 static
 sstables::shared_sstable create_sstable(sstables::test_env& env, schema_ptr s, std::vector<mutation> mutations) {
-    static thread_local auto tmp = tmpdir();
     static int gen = 0;
     return make_sstable_containing([&] {
-        return env.make_sstable(s, tmp.path().string(), gen++);
+        return env.make_sstable(s, gen++);
```

Comment:
nit: You can drop gen++ too, sstables::test_env maintains monotonically incrementing generation on its own